validators/cgroup_validator.go

```diff
@@ -82,6 +103,21 @@ func (c *CgroupsValidator) validateCgroupSubsystems(cgroups, subsystems []string
 }
 
 func (c *CgroupsValidator) getCgroupSubsystems() ([]string, error) {
+	// /proc/cgroups is meaningless for v2
+	// https://github.com/torvalds/linux/blob/v5.3/Documentation/admin-guide/cgroup-v2.rst#deprecated-v1-core-features
+	if IsCgroup2UnifiedMode() {
+		// "pseudo" controllers do not appear in /sys/fs/cgroup/cgroup.controllers.
+		// - devices: implemented in kernel 4.15
+		// - freezer: implemented in kernel 5.2
```

Comment:
i think having this list of the two controllers needs a separate reference link explaining why they are picked.

Reply:
Recent kernel does not show all controllers in /sys/fs/cgroup/cgroup.controllers. This line
https://github.com/torvalds/linux/blob/v5.3/kernel/cgroup/cgroup.c#L433-L434 actually decides what cgroup subsystems are masked. So "devices" and "freezer" are not visible.